packages/openapi-generator/src/file-serializer/schema.ts

```diff
@@ -32,13 +32,18 @@ export function serializeSchema(schema: OpenApiSchema): string {
     const type = serializeSchema(schema.items);
     return schema.uniqueItems
       ? `Set<${type}>`
-      : 'properties' in schema.items
+      : 'properties' in schema.items || 'allOf' in schema.items
```

Comment:
[req] This only solves it for `allOf`, but we should make sure this is also fixed for `oneOf` and `anyOf`.